client/data/settings/hooks.js

```diff
@@ -10,6 +11,9 @@ import { useSelect, useDispatch } from '@wordpress/data';
  */
 import { STORE_NAME } from '../constants';
 
+/**
+ * @return {import('../../types/wcpay-data-hooks').GenericSettingsHook<boolean>}
```

Comment:
I'm using this syntax (instead of converting this whole file in TS) because there still are some hooks that are a bit iffy.
However, these changes should get us 98% of the way there, without having to invest additional time.